var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -1,48 +1,31 @@
-##############################################################################
-# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
-# Produced at the Lawrence Livermore National Laboratory.
-#
-# This file is part of Spack.
-# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
-# LLNL-CODE-647188
-#
-# For details, see https://github.com/spack/spack
-# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
-#
-# This program is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License (as
-# published by the Free Software Foundation) version 2.1, February 1999.
-#
-# This program is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
-# conditions of the GNU Lesser General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public
-# License along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
-##############################################################################
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
 from spack import *
-from spack.pkg.builtin.neurodamus_base import NeurodamusBase
-import os
+from contextlib import contextmanager
 import shutil
+import os
 import sys
-from contextlib import contextmanager
 
 
-class Neurodamus(NeurodamusBase):
-    """Package used for building special from NeurodamusBase package
+class NeurodamusModel(Package):
+    """An 'abstract' base package for Simulation Models. Therefore no version.
+       Eventually in the future Models are independent entities, not tied to neurodamus
     """
-    variant('coreneuron', default=True, description="Enable CoreNEURON Support")
-    variant('profile', default=False, description="Enable profiling using Tau")
-    variant('python', default=False, description="Enable Python Neurodamus")
-    variant('syntool', default=True, description="Enable Synapsetool reader")
-    variant('sonata', default=False, description="Enable Synapsetool with Sonata")
+    depends_on('neurodamus-core')
+
+    variant('coreneuron', default=False, description="Enable CoreNEURON Support")
+    variant('profile',    default=False, description="Enable profiling using Tau")
+    variant('python',     default=False, description="Enable Python Neurodamus")
+    variant('syntool',    default=True,  description="Enable Synapsetool reader")
+    variant('sonata',     default=False, description="Enable Synapsetool with Sonata")
+    variant('debug',      default=False, description="Compile in debug mode (-O0)")
```

Comment:
Now I think we should remove debug variant here and just use `spack install neurodamus ^neuron+debug` whenever it requires. What you think?

Reply:
I introduced it recently because I thought it could make sense to build also the mods as debug. But I'm confused now since I saw mod files being compiled in debug just by using debug Neuron. Whats the magic there? In that case maybe we can drop this yes.

Reply:
Clarified offline. We can totally drop this option since mod comfilation flags are brought from neuron